packages/azure-kusto-data/src/security.ts

```diff
@@ -37,11 +40,13 @@ export class AadHelper {
         } else if (kcsb.accessToken) {
             this.tokenProvider = new BasicTokenProvider(kcsb.dataSource, kcsb.accessToken as string);
         } else if (kcsb.useUserPromptAuth) {
-            this.tokenProvider = new UserPromptProvider(kcsb.dataSource, kcsb.interactiveCredentialOptions);
+            this.tokenProvider = new UserPromptProvider(kcsb.dataSource, kcsb.interactiveCredentialOptions, kcsb.timeoutMs);
         } else if (kcsb.tokenProvider) {
             this.tokenProvider = new CallbackTokenProvider(kcsb.dataSource, kcsb.tokenProvider);
         } else if (kcsb.useDeviceCodeAuth) {
-            this.tokenProvider = new TokenProvider.DeviceLoginTokenProvider(kcsb.dataSource, kcsb.deviceCodeCallback, kcsb.authorityId);
+            this.tokenProvider = new TokenProvider.DeviceLoginTokenProvider(kcsb.dataSource, kcsb.deviceCodeCallback, kcsb.authorityId, kcsb.timeoutMs);
+        } else if (kcsb.tokenCredential) {
```

Comment:
Shouldn't this also include a "withTokenCredential" ctor?

Reply:
i dont get this comment